src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -138,17 +138,17 @@
                               __LINE__,                     \
                               FCNAME,                       \
                               fi_strerror(-_ret));          \
-        MPIR_ERR_CHKANDJUMP(_retry == 0,                    \
-                            mpi_errno,                      \
-                            MPIX_ERR_EAGAIN,                \
-                            "**eagain");                    \
         if (LOCK == MPIDI_OFI_CALL_NO_LOCK)                 \
             MPID_THREAD_CS_EXIT(POBJ,MPIDI_OFI_THREAD_FI_MUTEX);     \
         MPIDI_OFI_PROGRESS_NONINLINE();                              \
         if (LOCK == MPIDI_OFI_CALL_NO_LOCK)                 \
             MPID_THREAD_CS_ENTER(POBJ,MPIDI_OFI_THREAD_FI_MUTEX);    \
-        if (EAGAIN) _retry--;                                        \
-    } while (_ret == -FI_EAGAIN);                           \
+        _retry--;                                           \
+    } while (_retry != -1 || !EAGAIN);                           \
+    MPIR_ERR_CHKANDJUMP(_ret == -FI_EAGAIN && EAGAIN,       \
```

Comment:
This is moving a branch that was previously protected by `if (likely(_ret==0)) break;`. It may now be more penalizing because of that.

Reply:
Ah, that's true. I should move this `MPIR_ERR_CHKANDJUMP` back to its original place. So I guess the cost of it is avoidable in the retry loop.